Make all heading levels consistent

jirijakes opened the pull request:
With this change, all documents have a single level-1 heading (document title), all other headings are at least level 2.

Additionally, linter is introduced to enforce the heading consistency.

#### Rationale

I wish to export documents to other formats (using pandoc). Unfortunately, levels of headings were used inconsistently across the documents, which made automated export quite complicated. This change makes all headings consistent with 00-introduction.md and 01-messaging.md.

Headings were shifted down by one level, a few did not require any change. Links to sections remain the same, therefore no references to the bolts and their sections should be broken.

jirijakes committed:
Make all heading levels consistent

Previously, levels of headings were used inconsistently across the
documents, which made automated export to other formats complicated.
With this change, all documents have single level 1 heading (title of
the document) and all other headings are at least level 2.

ProofOfKeags commented:
ACK a59746d

ProofOfKeags commented:
Concept ACK

Seems like you should also specify this requirement in the [CONTRIBUTING.md](https://github.com/lightning/bolts/blob/master/CONTRIBUTING.md)

Also this is the type of thing you probably need a GitHub workflow to enforce, it's hard to review PR's at a glance and ensure they meet this requirement

jirijakes commented:
> Seems like you should also specify this requirement in the CONTRIBUTING.md
> Also this is the type of thing you probably need a GitHub workflow to enforce

Thanks, @ProofOfKeags , good ideas. I will look into it.

vincenzopalazzo reviewed (commented):
Concept ACK https://github.com/lightning/bolts/pull/1080/commits/a59746d1efe67b46818d7d7ed23552ec5501b89e

jirijakes committed:
Add markdown linting to GitHub workflow

Commit introduces initial set of rules for checking formatting of
markdown files as part of GitHub workflow. The rules are:

  - first line is top-level heading
  - there is only one top-level heading in the document

jirijakes commented:
Changes:

  - rebased
  - added to style guide
  - added linting workflow + simple script
    - uses https://github.com/DavidAnson/markdownlint-cli2
    - checks only what this PR is solving, other checks might be added gradually
    - tested on https://github.com/jirijakes/bolts/actions/runs/8026298250/job/21928515820

ProofOfKeags reviewed (approved):
ACK. Thanks for the cleanup and infrastructure to maintain it 🪴

vincenzopalazzo reviewed (approved):
ACK https://github.com/lightning/bolts/pull/1080/commits/c1f8a9d05180da4f254cdfe32ad258d99943a079

jirijakes commented:
Rebased.

Linter run: https://github.com/jirijakes/bolts/actions/runs/9318710058/job/25651599958

jirijakes commented:
Rebased and included BOLT12.

Linter run: https://github.com/jirijakes/bolts/actions/runs/11215057123/job/31171445815
